### PR TITLE
[PyTorch] add test for numel slow path affecting data_ptr

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100993

This test would have stopped #98090 -- data_ptr needs to call custom Python numel if it exists, since it could be arbitrary Python.

Differential Revision: [D45701566](https://our.internmc.facebook.com/intern/diff/D45701566/)

### CODE REVIEW

**Comment by swolchok** [comment]
hm, I'm not sure why this test doesn't pass everywhere.

File: test/test_python_dispatch.py [hunk 1]
```diff
@@ -2306,6 +2306,28 @@ def __torch_dispatch__(cls, func, types, args, kwargs):
             e = SizesDefaultReturn(torch.randn(4, 2), use_wrapper_subclass)
             self.assertEqual(e.size(), (4, 2))
 
+    def test_data_ptr_respects_numel_slow_path(self):
+        data = torch.randn(6, 2)
+        numel_called = [False]
```
Comment (ezyang) [inline test/test_python_dispatch.py:2311]: this is dead

**Comment by ezyang** [comment]
idk, it kind of looks like it is failing everywhere it is run

**Comment by swolchok** [comment]
> idk, it kind of looks like it is failing everywhere it is run

weird, passed locally and on internal CI

**Comment by swolchok** [comment]
@ezyang is there something internally enabling sym_numel hooking? I don't see any other tests that the functionality even works

**Comment by ezyang** [comment]
If you're talking fbcode, this might be that macro where we disabled TensorImpl extensibility for perf. Would need an OSS build to trigger then.

**Comment by swolchok** [comment]
maybe it needs a rebase to make sure we're atop an OSS commit that has the revert.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**